### PR TITLE
feat(linter): add rule `eslint/new-cap` 

### PR DESCRIPTION
**Pull request by Sysix**
https://eslint.org/docs/latest/rules/new-cap

Same Problem with rust regex instead of js regex. Added a note in the docs

### CODE REVIEW

File: crates/oxc_linter/src/rules/eslint/new_cap.rs [hunk 1]
```diff
@@ -0,0 +1,800 @@
...
+            None,
+        ),
+        ("var a = new c();", None),
+        ("var a = new b[ ( 'foo' ) ]();", None), // { "ecmaVersion": 6 },
+        ("var a = new b[`foo`];", None),         // { "ecmaVersion": 6 },
+        // (
+        //     "var a = b[`\\
+        // 	Foo`]();",
+        //     None,
+        // ), // { "ecmaVersion": 6 },
```
Comment (Sysix) [inline crates/oxc_linter/src/rules/eslint/new_cap.rs:783]: I do not know why this should fail. we are ignoring non alpha character (`GetCapResult`)

File: crates/oxc_linter/src/rules/eslint/new_cap.rs [hunk 2]
```diff
@@ -0,0 +1,795 @@
...
+}
+
+fn bool_serde_value(map: &serde_json::Map<String, serde_json::Value>, key: &str) -> bool {
+    let Some(value) = map.get(key) else {
+        return true; // default value
+    };
+
+    let err = format!("eslint/new-cap: expect configuration option '{key}' to be a boolean.");
+
+    value.as_bool().expect(&err)
```
Comment (camc314) [inline crates/oxc_linter/src/rules/eslint/new_cap.rs:50]: we typically `unwrap_or_default` in these cases to vaoid panics.

Comment (Sysix) [reply crates/oxc_linter/src/rules/eslint/new_cap.rs]: I found some rules which are already doing this :)

Removed it with https://github.com/oxc-project/oxc/pull/8146/commits/b8a83e517b5d6c848555bbf05aea830cee5c77e3

**Review by camc314 — APPROVED** [review]

File: crates/oxc_linter/src/rules/eslint/new_cap.rs (hunk 2 above)
Comment (camc314) [reply crates/oxc_linter/src/rules/eslint/new_cap.rs]: 😬

**Pull request merged**